src/types.ts

```diff
@@ -11,21 +16,40 @@ interface Refactoring {
   };
 }
 
+interface ActionProvider {
+  message: string;
+  isPreferred?: boolean;
+  createVisitor: (
+    selection: Selection,
+    onMatch: (path: NodePath<any>) => void,
```

Comment:
Doesn't `NodePath` has a default value for the generics?

```suggestion
    onMatch: (path: NodePath) => void,
```